src/core/queryInstance.ts

```diff
@@ -0,0 +1,104 @@
+import { uid, isServer, isDocumentVisible, Console } from './utils'
+import { Query, QueryState } from './query'
+import { BaseQueryConfig } from './types'
+
+// TYPES
+
+type OnStateUpdateFunction<TResult, TError> = (
+  state: QueryState<TResult, TError>
+) => void
+
+// CLASS
+
+export class QueryInstance<TResult, TError> {
+  id: number
+  onStateUpdate: OnStateUpdateFunction<TResult, TError>
+  config: BaseQueryConfig<TResult, TError>
+
+  private query: Query<TResult, TError>
+  private refetchIntervalId?: number
+
+  constructor(
+    query: Query<TResult, TError>,
+    onStateUpdate: OnStateUpdateFunction<TResult, TError>
+  ) {
+    this.id = uid()
+    this.onStateUpdate = onStateUpdate
+    this.query = query
+    this.config = {}
+  }
+
+  clearInterval(): void {
+    clearInterval(this.refetchIntervalId)
```

Comment:
Do we need a check wether `refetchIntervalId` is present before we clear it?

Reply:
Added checks before clearing